x-pack/plugins/security_solution/server/lib/telemetry/sender_v2.ts

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import axios from 'axios';
+import * as rx from 'rxjs';
+import { cloneDeep } from 'lodash';
+
+import type { Logger } from '@kbn/core/server';
+import type { TelemetryPluginSetup } from '@kbn/telemetry-plugin/server';
+import type { ITelemetryReceiver } from './receiver';
+import {
+  type ITelemetryEventsSenderV2,
+  type QueueConfig,
+  type RetryConfig,
+  type TelemetryEventSenderConfig,
+} from './sender_v2.types';
+import { type TelemetryChannel } from './types';
+import * as collections from './collections_helpers';
+import { CachedSubject, retryOnError$ } from './rxjs_helpers';
+import { SenderUtils } from './sender_helpers';
+import { newTelemetryLogger, type TelemetryLogger } from './helpers';
+
+export class TelemetryEventsSenderV2 implements ITelemetryEventsSenderV2 {
+  private retryConfig: RetryConfig | undefined;
+  private queues: Map<TelemetryChannel, QueueConfig> | undefined;
+
+  private readonly events$ = new rx.Subject<Event>();
+
+  private readonly finished$ = new rx.Subject<void>();
+  private cache: CachedSubject<Event> | undefined;
+
+  private status: ServiceStatus = ServiceStatus.CREATED;
+
+  private readonly logger: TelemetryLogger;
+
+  private senderUtils: SenderUtils | undefined;
+
+  constructor(logger: Logger) {
+    this.logger = newTelemetryLogger(logger.get('telemetry_events'));
+  }
+
+  public setup(
```

Comment:
You may have just not gotten to it yet, since this is still in draft pr status, but just a heads up to call setup and start in [plugin.ts](https://github.com/elastic/kibana/blob/e7927469f653926d23e8f2f7037bcaa111c88243/x-pack/plugins/security_solution/server/plugin.ts#L448)